chapters/modules/chapter_8/pages/sharp.adoc

```diff
@@ -0,0 +1,78 @@
+
+At first glance, recursive proofs may seem more complex and time-consuming. However, there are several benefits to this approach:
+
+. *Parallelization*: Recursive proofs allow for work parallelization, reducing user latency and improving SHARP efficiency.
+. *Lower cloud costs*: Since each job is shorter, the required memory for processing is reduced, resulting in lower cloud costs.
+. *Optimization*: Recursive proofs enable the SHARP to optimize for various factors, including latency, on-chain costs, and time to proof.
+
+[NOTE]
+====
+Latency in Starknet encompasses the time taken for processing, confirming, and including transactions in a block. It is affected by factors like network congestion, transaction fees, and system efficiency. Minimizing latency ensures faster transaction processing and user feedback.
+
+Time to proof, however, specifically pertains to the duration required to generate and verify cryptographic proofs for transactions or operations.
+====
+
+== SHARP Backend Architecture and Data Pipeline
+
+The SHARP back end architecture consists of several services that work together to process Cairo jobs and generate proofs. These services include:
+
+. *Gateway*: Cairo jobs enter the SHARP through the gateway.
+. *Validator*: This is the first important step. The validator service runs validation checks on each job, ensuring they meet the requirements and can fit within the prover machines. Invalid jobs are tagged as such and do not proceed to the Prover.
+. *Scheduler*: The scheduler service creates "trains" that aggregate jobs and send them to the Prover. Recursive jobs are paired and sent to the Prover together.
+. *Cairo Runner*: This service runs Cairo for the Prover’s needs. The Cairo Runner service runs Cairo programs, executing the necessary computations and generating the execution trace as an intermediate result. The Prover then uses this execution trace.
+. *Prover*: The Prover computes the proofs for the jobs.
+. *Dispatcher*: The dispatcher sends the proofs to either the Cairo Verifier for further recursion or the Solidity Verifier on Ethereum for final verification.
+
+The SHARP is designed to be stateless (each Cairo job is executed in its own context and has no dependency on other jobs), allowing for greater flexibility in processing jobs. This allows greater flexibility when processing jobs. It also relies on the cryptographic soundness of the Verifier, eliminating the need to trust the Prover’s computations.
+
+== Current SHARP Users
+
+Currently, the primary users of SHARP include:
+
+* StarkEx
+* Starknet
+* External users who use the Cairo Playground
+
+== Challenges and Optimization
+
+Optimizing the Prover involves numerous challenges and potential projects on which the Starkware team and the community are currently working:
```

Comment:
We will also, probably in the future (might take some time :) ), improve the scheduling algorithm, that's also an optimisation path we can take. It is not within the prover itself though.